assets/css/components/_tag.scss

```diff
@@ -9,6 +9,7 @@
   transition: background 0.8s;
 
   &:hover {
+    transition: color 0.8s;
```

Comment:
It's not needed because you have already defined in `a` style definition. https://github.com/vaga/hugo-theme-m10c/blob/007be0a4dff4d2f8bebcd8295da198adc0a9bd32/assets/css/_base.scss#L22

Reply:
I've put the color transition of the `a` from `a:hover` to directly `a` so `.tag:hover` no longer inherits it. The change in `a` was needed to fix the zero fade-out time of `a` (urls in posts, and social links under the avatar) that looked very bad, especially combined with a very long fade in time.

I'm 100% sure that this transition setting her is needed to preserve the original "knight-rider" colour animation of the tags (try to quickly hover over the tags from left to right and vice-versa with and without this transition setting). I've actually played/experimented with this (and read some discussions about the reason) for about an hour to find out why this is needed and how it works and the answer lays in how the transition settings of `.tag` and `.tag:hover` are stored and used.

Reply:
You might be much more versed in CSS then I am but in order to avoid more roundtrips in this conversation I try to explain what's going on here (based on my experiments in Chrome when I made these changes):

**`transition` statements**
Setting the the transition only for `color` without specifying `background` automatically sets a zero `background` transition and vice versa:
- `transition color 1s;` sets 1s transition for color and 0s for background
- `transition background 1s;` sets 1s transition for background and 0s for color
- `transition color 1s, background 2s;` sets 1s transition for color and 2s for background

In case of our `.tag`:
- `.tag` transition times: color=0s background=0.8s (because of `transition background 0.8s;`)
- `.tag:hover` transition times: color=0.8s background=0s (because of `transition color 0.8s;`)

The `.tag.hover` transition settings are used only when the cursor is over the tag while `.tag` transitions settings are used only when the cursor isn't over the tag. This means the following:
- tag text fade-in time: 0.8s (defined in `.tag:hover`)
- background fade-in time: 0s (defined in `.tag:hover`)
- tag text fade-out time: 0s (defined in `.tag`)
- background fade-out: 0.8s (defined in `.tag`)

In this pull request my changes preserve the above settings while in my custom project-specific CSS I use fade-in=0.1s and fade-out=0.35s both for color and background. Having shorter (but a bit longer than zero) anims generally look better and more pro in my opinion. In my experience it's better to keep UI animation (both movement and color) in the 0.05s-0.5s range.

**inheriting transitions**
(again, based on my experiments in Chrome)

The `color` and `background` transitions are inherited together, it seems to be one setting. Now let's see a few examples of how inheriting works in our example where we have 4 actors: `a`, `a:hover`, `.tag` and `.tag:hover`.

- If we set transition only for `a` then all 4 actors inherit the same setting.
- If we set `a` and `a:hover` then `.tag` inherits from `a` and `.tag:hover` inherits from `a:hover`
- If we set `a` and `.tag` then `a:hover` inherits from `a` while `.tag:hover` inherits from `.tag`
- If we set `a:hover` and `.tag` then `.tag:hover` inherits from `a:hover`

Based on the above observations: we can put the `transition color 0.8s` either into `a:hover` (while still having `transition color 0.8s` directly in `a` too to have proper `a` fade-out) or `.tag:hover` to achieve the same tag color animations but I decided to put it into `.tag:hover` because this way other classes that inherit from `a` have to deal only with one transition setting in `a` instead of 2 transition settings in `a` and `a:hover` when it comes to inheriting and overriding settings. And this hover transition setting belongs to `.tag` anyway, `a` needs only the transition that is directly in `a`.

This bacame pretty long. Sorry for this spaghetti message. :-D

Reply:
Wow! Your explanations is very detailed. Thanks!

I'm not an UI expert and you have definitely right about UI animation delay.
You can add your custom animation settings in this request to replace current one. I think it was better for all of us. :-)

Reply:
I have some UI design related experience from game programming and UI framework programming. This CSS transition thingy is a new territory for me but my "OCD" didn't let me stop debugging/fixing some of these issues. :-D

I've added two commits:

1. In the first one I fixed the zero fade-out time of the tag text color (by keeping the rest of the transition settings).
2. In the second commit I set the `a` and `.tag` transition times to what I'm using in my project. 

This way you can easily compare them and decide which one you like more. To some degree it's a matter of personal preference. If you prefer the original settings then we simply remove the last commit.

Reply:
A jsfiddle where the double-monospace issue can be tried along with the tag transition settings: https://jsfiddle.net/okz69es7/